apps/www/components/Discussion/overlays/DiscussionPreferences.js

```diff
@@ -356,6 +357,11 @@ const DiscussionPreferencesEditor = ({
                   setState((curr) => ({
                     ...curr,
                     anonymity: val,
+                    credential:
+                      val === false // if not anonymous set credential
+                        ? userPreference?.credential?.description ??
+                          autoCredential?.description
+                        : '',
```

Comment:
Let's keep credential field value when unselecting anonymous:
```suggestion
                      val === false // if not anonymous set credential
                        ? state.credential || userPreference?.credential?.description || autoCredential?.description || ''
                        : '',
```

Also note that the defaulting to an empty string is necessary for the input field to update correctly:
<img width="1542" alt="Screenshot 2022-05-25 at 00 05 45" src="https://user-images.githubusercontent.com/410211/170141198-4704bebf-0804-47d9-b7c9-6d5efa4efa47.png">
(this happens when setting it to undefined)